MoreLinq/ExceptBy.cs

```diff
@@ -150,11 +154,10 @@ static partial class MoreEnumerable {
             var keys = new HashSet<TKey>(second, keyComparer);
             foreach (var element in first) {
                 var key = keySelector(element);
-                if (keys.Contains(key)) {
-                    continue;
-                }
-                yield return element;
-                keys.Add(key);
+                if (!keys.Contains(key)) {
+                    yield return element;
+                    keys.Add(key);
```

Comment:
[Add returns a boolean indicating whether the element was already present](https://msdn.microsoft.com/en-us/library/bb353005(v=vs.110).aspx) So no need to invoke both Contains and Add